File: frontend/src/lib/api/types/alert.ts
import type { Pagination } from "./api";

/**
 * Represents a single site alert.
 */
export interface SiteAlert {
  alert_id: string;
  title: string;
  body: string;
  start_time: string;
  end_time: string;
  is_active: boolean;
  creator_name: string;
  created_at: string;
  updated_at: string;
}

/**
 * Payload for creating or updating a site alert.
 */
export interface SiteAlertPayload {
  title: string;
  body: string;
  start_time: Date | string;
  end_time: Date | string;
  is_active: boolean;
}

/**
 * Paginated response for a list of site alerts.
 */
export interface PaginatedAlertsResponse {
  alerts: SiteAlert[];
  pagination: Pagination;
}

File: backend/internal/models/api/alert.go
package models

import "time"

// --- Request Payloads ---

// Payload for creating or updating a site alert
type SiteAlertRequest struct {
	Title     string    `json:"title" validate:"required"`
	Body      string    `json:"body" validate:"required"`
	StartTime time.Time `json:"start_time" validate:"required"`
	EndTime   time.Time `json:"end_time" validate:"required"`
	IsActive  bool      `json:"is_active"`
}

// --- Response Payloads ---

// Represents a single site alert in an API response
type SiteAlertResponse struct {
	AlertID     string    `json:"alert_id"`
	Title       string    `json:"title"`
	Body        string    `json:"body"`
	StartTime   time.Time `json:"start_time"`
	EndTime     time.Time `json:"end_time"`
	IsActive    bool      `json:"is_active"`
	CreatorName string    `json:"creator_name"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// Paginated response for a list of site alerts
type PaginatedAlertsResponse struct {
	Alerts     []SiteAlertResponse `json:"alerts"`
	Pagination Pagination          `json:"pagination"`
}
